gems/aws-sdk-core/lib/aws-sdk-core/client_stubs.rb

```diff
@@ -17,6 +20,20 @@ def setup_stubbing
           apply_stubs(operation_name, Array === stubs ? stubs : [stubs])
         end
       end
+
+      # When a client is stubbed allow the user to access the requests made
+      @api_requests = []
+
+      requests = @api_requests
+      self.handle do |context|
+        requests << {
+          metadata: context.metadata,
+          request: context.http_request,
+          response: context.http_response,
+          context: context
+        }
```

Comment:
I think it would make sense to explicitly pull out and provide the params. See: https://github.com/aws/aws-sdk-ruby-record/blob/master/spec/aws-record/record/item_operations_spec.rb#L45

Consider pulling out `context.params` as the member `:params`, `context.operation.name` as the member `:operation`, and then just dropping the `:context` object. I think that will focus on the most important parts, while still providing the full context.